### ISSUE TITLE
[17.0] [hr_attendance] Kiosk mode cannot be opened

### ISSUE THREAD

**Issue by feelwhy** [odoo/odoo#144302]
Impacted versions:
17.0 (both Enterprise and Community; demo.odoo.com as well)

Steps to reproduce:
1. Install the module "Attendances" (hr_attendance)
2. Make sure you are in multi-company mode (so there are a few companies available for your user)
3. Go to Attendances > Kiosk Mode

Current behavior:
500: Internal Server Error
In the logs:
```
Traceback (most recent call last):
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/http.py", line 2157, in __call__
    response = request._serve_db()
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/http.py", line 1732, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/http.py", line 1759, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/http.py", line 1873, in dispatch
    return self.request.registry['ir.http']._dispatch(endpoint)
  File "/home/feelwhy/Odoo/Odoo17/odoo/addons/website/models/ir_http.py", line 235, in _dispatch
    response = super()._dispatch(endpoint)
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/addons/base/models/ir_http.py", line 207, in _dispatch
    result = endpoint(**request.params)
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/http.py", line 722, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/home/feelwhy/Odoo/Odoo17/odoo/addons/hr_attendance/controllers/main.py", line 83, in open_kiosk_mode
    } for e in request.env['hr.employee'].sudo().search_read(domain=[('company_id', '=', company.id)],
  File "/home/feelwhy/Odoo/Odoo17/odoo/odoo/fields.py", line 5118, in __get__
    raise ValueError("Expected singleton: %s" % record)
ValueError: Expected singleton: res.company(2, 1)
```

Actually, the reason is that _get_company might return a few companies while in the domain there should be a single one. Probably, there is a problem with a passed token or just the search should be done for a few companies.

Expected behavior:
No error, it should be possible to register attendance in the kiosk mode.

**Comment by michaelhofer-slg** [comment]
I have just noticed the same issue. 
Did you ever find a workaround or fix for this?

**Comment by michaelhofer-slg** [comment]
> I have just noticed the same issue. Did you ever find a workaround or fix for this?

As it happens, I have just found a workround for this myself. It seems that I had multiple companies enabled at some point. Regenerating the Kiosk URL for the `hr_attendance` module did fix the issue.

**Comment by michaelhofer-slg** [comment]
Just noticed that this seems to be fixed in master: https://github.com/odoo/odoo/commit/55a14c38742c20c42e2a19835f4cf2e53ef94d9f

**Comment by vava-odoo** [comment]
Thanks @feelwhy for reporting. Fixed in 17.0 with https://github.com/odoo/odoo/pull/145382
